### ISSUE TITLE
Suppress output other than plot with RCall.ijulia_setdevice

### ISSUE THREAD

**Issue by dmbates** [JuliaInterop/RCall.jl#450]
In an IJulia notebook where I have set an R graphics device when I run a cell that creates an R plot the plot is displayed followed by the single output cell containing `RObject{VecSxp}`.  I would like to have the plot displayed without the output cell but I haven't figured out how to do that.  @simonbyrne can you suggest a way to do this? I am willing to work on a PR if I know where to make the change.

This would be very helpful when using [quarto](https://quarto.org).  At present in a Julia-based Quarto notebook, which uses Jupyter as the evaluation engine, the presence of the two pieces of output messes with the Figure captions.

**Comment by simonbyrne** [comment]
Does using a semicolon after the final expression work?

**Comment by dmbates** [comment]
> Does using a semicolon after the final expression work?

In the sense that it suppresses the printed output, yes. However, it also suppresses the plot display. 🙄

**Comment by simonbyrne** [comment]
honestly, it's been so long since I looked at the code that I've forgotten how it works...

Are you using ggplot2 or regular Julia plots?

**Comment by simonbyrne** [comment]
If you're using ggplot2 (which it sounds like, as suppressing the output stops the picture from displaying), then it's called as part of the `rprint` function:
https://github.com/JuliaInterop/RCall.jl/blob/2da13513856f7d82cff9c5ce8c6cb99c21c3631e/src/io.jl#L43

It might be possible to modify it in some way so that it doesn't print anything if the plot is displayed?

**Comment by dmbates** [comment]
In fact I am using `{lattice}` for the plots but I think the effect is the same as with `{ggplot2}`.  I had rather suspected that this would be where the modification would need to be made so I will take a look there.

**Comment by dmbates** [comment]
I tried explicitly returning `nothing` from the `ijulia_displayfile` and `ijulia_displayplots` methods but that unsurprisingly doesn't change the output.  I think it may be necessary to take a sledgehammer to the `rprint` method you linked to and do the checks for `isdefined(Main, :IJulia)`, etc. at the beginning of the function and the checking for `rcopy(Int, rcall_p(Symbol("dev.cur"))) != 1` there.  If there are plots to display in Jupyter then just display the plots and skip the rest of the `rprint` method.  As I said, it is kind of a sledgehammer but it's the best I can come up with at the moment.

**Comment by dmbates** [comment]
IIRC both ggplot2 and lattice return objects for which the plot output is displayed when R's `print` method is invoked, as is mentioned in the comments in the `rprint` function you linked to.  It seems that the call to `println(io, typeof(r))` in method `show(io::IO, r::RObject)` a few lines below is where the line we want to suppress is coming from.  The problem is that this line is executed before the call to `rprint(io, r)` and it is only at the end of the call to Julia's `rprint` method, during which R's `print` method is called in some way, that we discover if we displayed plots.  So we discover that plots have been created and displayed too late to suppress the printing of the offending `RObject{VecSxp}` line. I tried suppressing just that one `println` call in `show(io::IO, r::RObject)` if IJulia is defined and inited but it still produces a blank line in a Jupyter notebook and `quarto` gets confused.  I think I will ask on the quarto-cli discussions for ideas.
